server/main.py

```diff
@@ -114,6 +115,14 @@ async def index(request):
     job.env('HAIL_BILLING_PROJECT', dataset)
     job.env('DATASET_GCP_PROJECT', dataset_gcp_project)
     job.env('OUTPUT', output_suffix)
+
+    if environment_variables:
+        for env_var_pair in environment_variables:
+            pair = env_var_pair.split('=')
```

Comment:
I think this formatting would be cleaner on the CLI (so keeping your original code here). You might want to consider the case where someone doesn't provide an equals on the CLI, or doesn't provide the map in the correct format.

For example:

```py
if not isinstance(environment_variables, dict):
    raise ValueError('Expected environment_variables to be dictionary')
invalid_env_vars = [f'{k}={v}' for k,v in environment_variables.items() if not isinstance(v, str)]
if len(invalid_env_vars) > 0:
    raise ValueError('Some values of the environment_Variables were not a string, got ' + ', '.join(invalid_env_vars))
```